### DISCUSSION TITLE
User Authentication

### DISCUSSION THREAD

**Discussion by ninahauserberlin** [rstudio/learnr#682] [Q&A]
Hello team!

We love your package. As our tutorial is quite large, we want to minimize user access by building an authentication system around it. Is there a best practice to approach this? We are familiar with shinyauthr but are unsure how to connect it.

Thanks for your great work and all the best,
Nina

**Comment by gadenbuie** [comment]
[RStudio Connect](https://www.rstudio.com/products/connect/) is a paid product from RStudio that can serve learnr tutorials and Shiny apps with robust user access controls.

I think that similar access control can be achieved using [shinyproxy](https://www.shinyproxy.io/) but I haven't tried it and don't know for certain that it would work for your use case.

I've used [shinyauthr](https://paulc91.github.io/shinyauthr/) before in a Shiny app, but I don't think that will work for learnr since you need to be able to control the shiny server logic to make the UI conditional on successful authentication.
